### ISSUE TITLE
$additionalProviderConfig is ignored when trying to override provider config with setConfig()

### ISSUE THREAD

**Issue by johnRivs** [SocialiteProviders/Manager#117]
# Issue / Motivation:
I'm following the instructions in this repo to override the provider config. I'm using the Spotify provider and I want to pass `['show_dialog' => true]` but when I see the URL the user is being redirected to, there's no `show_dialog`.

## Expected behavior:
The URL should contain the additional provider config I pass as a 4th parameter to `\SocialiteProviders\Manager\Config`.

## Actual behavior:
Everything except the additional provider config gets taken into consideration.

## Steps to reproduce the behavior:
Simply following the instructions in this repo's README, under Dynamically Passing a Config.

## Proposed solution:
I wish this would've been written like Taylor Otwell designed it initially:
```php
Socialite::driver('spotify')
    ->with($additionalParameters)
    ->redirect();
```

**Comment by johnRivs** [comment]
TL;DR: Use the first example in this comment and use strings instead of booleans.
---

I've had a look at the source code and learned a couple of things.

## Usage
I actually can write it like the end of the OP. When I use this:
```php
return Socialite::driver('spotify')
                ->with(['show_dialog' => true])
                ->scopes(Session::$scopes)
                ->redirect();
```
this is the object I'm dealing with:
![](https://i.imgur.com/S7U6XwV.png)
That works fine, however, when I use the example in the README:
```php
$credentials = config('services.spotify');

$config = new Config(
    $credentials['client_id'], 
    $credentials['client_secret'], 
    $credentials['redirect'], 
    ['show_dialog' => true]
);

return Socialite::with('spotify')
                ->setConfig($config)
                ->scopes(Session::$scopes)
                ->redirect();
```
the config is arranged slightly differently:
![](https://i.imgur.com/fxl9TG3.png)
Notice how `['show_dialog' => true]` is now in `config` instead of `parameters`.

It's the first time I interact with this package so I'm not really sure what is intended and what is not, so I don't want to submit anything because I don't know what implications it would have for other providers.

## Parameters parsing
Even though I got it acknowledge the additional parameters I still couldn't get Spotify to force authentication. Looking at it a bit closer I found out that [this line](https://github.com/laravel/socialite/blob/3.0/src/Two/AbstractProvider.php#L167) from Laravel Socialite will turn `'show_dialog=true'` into `'show_dialog=1'` because I was passing it as a boolean instead of a string. So if anybody reading this issue comes across this problem, just use strings.
